lib/tarpon/entity/offering.rb

```diff
@@ -0,0 +1,25 @@
+# frozen_string_literal: true
+
+module Tarpon
+  module Entity
+    Package = Struct.new(:identifier, :platform_identifier)
+    class Offering
+      attr_reader :identifier, :description, :packages
+
+      def self.from_json(json)
```

Comment:
This is misleading as the method expects a hash.

What about having keyword arguments in the constructor and having this logic there? As a bonus we get nicer error messages if an invalid hash is passed (missing mandatory fields for instance).

Reply:
Ok, agree on the naming part. Though what do you mean keyword arguments? if `from(description:, packages:, identifier)` then it kinda defeats the purpose. If there is a Ruby idiom for instantiating objects from hash, then I am happy to use it, otherwise I suggest only to rename it

Reply:
What purpose are you talking about?

My suggestion is having the keyword arguments in the constructor since all offering fields are required, which is achieved naturally by using keyword arguments. I just don't see the advantage of having this extra method to wrap the object construction.

But it's not a showstopper, my main point still is the method name, which I can't think of a suitable name.

Reply:
Having that in the constructor as I suggested before is another possible solution.

Reply:
Purpose is to have a factory that instantiates object from hash